go/libraries/doltcore/sqle/dprocedures/dolt_checkout.go

```diff
@@ -35,57 +35,108 @@ var ErrEmptyBranchName = errors.New("error: cannot checkout empty string")
 
 // doltCheckout is the stored procedure version for the CLI command `dolt checkout`.
 func doltCheckout(ctx *sql.Context, args ...string) (sql.RowIter, error) {
-	res, err := doDoltCheckout(ctx, args)
+	res, message, err := doDoltCheckout(ctx, args)
 	if err != nil {
 		return nil, err
 	}
-	return rowToIter(int64(res)), nil
+	return rowToIter(int64(res), message), nil
 }
 
-func doDoltCheckout(ctx *sql.Context, args []string) (int, error) {
+func generateSuccessMessage(newBranch, upstream string) string {
+	result := fmt.Sprintf("Switched to branch '%s'", newBranch)
+	if upstream != "" {
+		result += fmt.Sprintf("\nbranch '%s' set up to track '%s'.", newBranch, upstream)
+	}
+	return result
+}
+
+// willModifyDb determines whether or not this operation is a no-op and can return early with a helpful message.
```

Comment:
This is a helper function for a helper function. Push lower in the file. Keep the highest abstractions at the top, when possible. Non-blocking

Reply:
Moved into `dolt_checkout_helpers.go`